refl1d/probe/probe.py

```diff
@@ -1395,14 +1396,14 @@ def __init__(
         self.R = R
         self.dR = dR
         self.unique_L = None
-        self.calc_Qo = self.Q
+        self._calc_Q = np.unique(self.Q)
         self.name = name
         self.filename = filename
         self.resolution = resolution
 
     @property
     def calc_Q(self):
-        return self.calc_Qo if not self.back_reflectivity else -self.calc_Qo
+        return self._calc_Q if not self.back_reflectivity else -self._calc_Q
```

Comment:
Did you check whether the resolution calculation can handle decreasing instead of increasing Q values? Or is resolution doing its own sort just to be safe.

Reply:
I'm not sure.  That particular logic has been in place for a long time (returning -calc_Q for back_reflectivity=True) and hasn't changed with this PR - but we should definitely check it!

Reply:
Yes, in `BaseProbe.apply_beam` the calc_Q is negated after the reflectivity is calculated if back_reflectivity is True before running the resolution correction: https://github.com/reflectometry/refl1d/blob/master/refl1d/probe/probe.py#L256